src/Gulpfile.js

```diff
@@ -69,6 +88,12 @@ function resolveAssetGroupPaths(assetGroup, assetManifestPath) {
     assetGroup.inputPaths = assetGroup.inputs.map(function (inputPath) {
         return path.join(assetGroup.basePath, inputPath);
     });
+    
+    assetGroup.watchPaths = !!assetGroup.watch ?
+        assetGroup.watch.map(function (watchPath) {
+            return path.join(assetGroup.basePath, watchPath);
+        }).concat(assetGroup.inputPaths) : assetGroup.inputPaths;
```

Comment:
I still think this is the wrong place for this - the merging of input paths with watch paths should go in the "watch" task function, not here. The `watchPaths` property should retain its meaning and value, not be "magically" altered here to mean something different.